File: 0268-missing-number/0268-missing-number.cpp

class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum_1=0;
        // for(int i=0;i<=nums.size();i++){
        //     sum_1^=i;
        // }
        for(int i=0;i<nums.size();i++){
            sum_1^=nums[i];
            sum_1^=i;
            
        }
        return sum_1^nums.size();
       
        
    }
};

File: 0268-missing-number/0268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    const n=nums.length;
    // let frq=new Set();
    // for(let ele of nums){
    //     frq.add(ele);
    // }
    // for(let i=0;i<=n;i++){
    //     if(!frq.has(i))return i;
    //}
    
    let ans=0;
    for(let ele of nums){
        ans^=ele;
    }
    for(let i=0;i<=n;i++){
        ans^=i;
    }
    
    return ans;
};